src/bots/uncrossArbBot.ts

```diff
@@ -0,0 +1,242 @@
+	public readonly name: string;
+	public readonly dryRun: boolean;
+	public readonly defaultIntervalMs: number = 1000;
+
+	private driftEnv: DriftEnv;
+	private periodicTaskMutex = new Mutex();
+
+	private jitProxyClient: JitProxyClient;
+	private driftClient: DriftClient;
+	// private subaccountConfig: SubaccountConfig;
+	private intervalIds: Array<NodeJS.Timer> = [];
+
+	private watchdogTimerMutex = new Mutex();
+	private watchdogTimerLastPatTime = Date.now();
+
+	private dlobSubscriber: DLOBSubscriber;
+	private slotSubscriber: SlotSubscriber;
+	private userMap: UserMap;
+
+	constructor(
+		driftClient: DriftClient, // driftClient needs to have correct number of subaccounts listed
+		jitProxyClient: JitProxyClient,
+		slotSubscriber: SlotSubscriber,
+		config: BaseBotConfig,
+		driftEnv: DriftEnv
+	) {
+		this.jitProxyClient = jitProxyClient;
+		this.driftClient = driftClient;
+		this.name = config.botId;
+		this.dryRun = config.dryRun;
+		this.driftEnv = driftEnv;
+		this.slotSubscriber = slotSubscriber;
+
+		this.userMap = new UserMap(
+			this.driftClient,
+			this.driftClient.userAccountSubscriptionConfig,
+			false
+		);
+		this.dlobSubscriber = new DLOBSubscriber({
+			dlobSource: this.userMap,
+			slotSource: this.slotSubscriber,
+			updateFrequency: 1000,
+			driftClient: this.driftClient,
+		});
+	}
+
+	/**
+	 * Run initialization procedures for the bot.
+	 */
+	public async init(): Promise<void> {
+		logger.info(`${this.name} initing`);
+
+		await this.userMap.subscribe();
+		await this.dlobSubscriber.subscribe();
+
+		logger.info(`${this.name} init done`);
+	}
+
+	/**
+	 * Reset the bot - usually you will reset any periodic tasks here
+	 */
+	public async reset(): Promise<void> {
+		// reset any periodic tasks
+		for (const intervalId of this.intervalIds) {
+			clearInterval(intervalId as NodeJS.Timeout);
+		}
+		this.intervalIds = [];
+	}
+
+	public async startIntervalLoop(intervalMs: number): Promise<void> {
+		const intervalId = setInterval(
+			this.runPeriodicTasks.bind(this),
+			intervalMs
+		);
+		this.intervalIds.push(intervalId);
+
+		logger.info(`${this.name} Bot started! driftEnv: ${this.driftEnv}`);
+	}
+
+	/**
+	 * Typically used for monitoring liveness.
+	 * @returns true if bot is healthy, else false.
+	 */
+	public async healthCheck(): Promise<boolean> {
+		let healthy = false;
+		await this.watchdogTimerMutex.runExclusive(async () => {
+			healthy =
+				this.watchdogTimerLastPatTime > Date.now() - 2 * this.defaultIntervalMs;
+		});
+		return healthy;
+	}
+
+	/**
+	 * Typical bot loop that runs periodically and pats the watchdog timer on completion.
+	 *
+	 */
+	private async runPeriodicTasks() {
+		const start = Date.now();
+		let ran = false;
+		try {
+			await tryAcquire(this.periodicTaskMutex).runExclusive(async () => {
+				console.log(
+					`[${new Date().toISOString()}] Running uncross periodic tasks...`
+				);
+				const marketIndexes = PerpMarkets[this.driftEnv];
```

Comment:
can we just use getMarketAccounts on driftClient?

Reply:
done